### ISSUE TITLE
PSA: The repo has moved to submodules. What does it imply?

### ISSUE THREAD

**Issue by Tk-Glitch** [Tk-Glitch/PKGBUILDS#534]
Some may have noticed that the master branch is now archived. The new default branch, `frogging-family`, is now submodule-based. Each individual repo can now be found in a newly created organization: https://github.com/Frogging-Family

Pros:
 - Allows you to clone only the project(s) you're interested in if needed while keeping the ability to grab everything from PKGBUILDS.
 - Allows me to push WIP or experimental code without breaking the version available via PKGBUILDS.
 - Allows for actual, individual releases (and there's more to come regarding this).
 - Well separated commit logs, allowing for better tracking of what's going on in which project.
 - Allows me to finally give **you** more power regarding community-patches, without the PR overhead.

Cons:
 - The commit log of PKGBUILDS itself won't be as clear to follow due to additional overhead. I'll try to find the best way to give as detailed and clear logs as possible without making it unmanageable in the long run.
 - "Loss" of the previous index. It's still there in the archived master branch, but obviously past tracking will be much more annoying.

All in all I think this change is for the better.

**For people using my tools in their workflow, there are a few changes that might need adapting while using the full PKGBUILDS toolset:**
 - The `linux-tkg` kernels are now in their own unified dir/project, which then contains each current version available subdirs. It might affect paths on your end due to the additional level.
 - `Proton-tkg` and `wine-tkg-git` have been unified as well in the `wine-tkg-git` dir/project. It might affect paths on your end due to the additional level. Sub-hierarchy and usage is unchanged.

**Please let me know if you have questions, opinions or requests regarding this change. Thank you!**

**Comment by CartoonFan** [comment]
I noticed that each submodule has an issue tracker now. Do you want us to open issues in the relevant submodule or continue to put everything here at the "central" PKGBUILDS location?

Thanks @Tk-Glitch!

**Comment by Tk-Glitch** [comment]
Good question! I'm not against per-submodule, and it might be what makes the most sense at the end of the day, but it also breaks habits, be it for me or for you. Having everything centralized sounds more straightforward to handle for a single person, but I'll let you guys decide. I'll get your issues mailed to me either way 🐸

**Comment by CartoonFan** [comment]
That's good to know. I guess time and experience will help me figure out where to post things. It's somewhat off-topic, but since you're here, it might be a good place to ask about wine-tkg bugs. I see that the line
> PLEASE DO NOT REPORT BUGS ENCOUNTERED WITH THIS AT WINEHQ OR VALVESOFTWARE, REPORT HERE INSTEAD !

is still part of the wine-tkg readme. Despite the all-caps text and the exclamation point, I'm still a little unsure of the scope of what I should report. Should I report bugs that are already on WineHQ or Valve's bug trackers (I'm thinking probably not)? Are new Wine bugs that are not strictly TkG-related OK (I'm thinking probably yes, but again I'm not sure)? The main reason I ask is that it feels awkward to mention Wine-TkG versions on the WineHQ bug tracker, and it feels weird to bring it up here if there are already bugs elsewhere or if it's not TkG-patch related.

Again, sorry if it's off-topic, but I'm kind of curious and I haven't really seen it described in detail here yet. Thanks again!

**Comment by Tk-Glitch** [comment]
You should never report a bug encountered with a custom Wine build to winehq unless asked otherwise.
There are clean profiles available to you with wine-tkg (that you can enable with the `_LOCAL_PRESET` config option) so you can build clean mainline or staging builds, suitable for logging and reporting to winehq.

Regarding Proton it's a bit more delicate as we are pretty far away from Valve's Wine. Between the devel Wine, staging and rebased patches - and even with a minimal patch list - there are many moving parts that can (and do) break every day. So it can be considered totally unsuitable for reporting to Valve.
You *could* technically get close by using Valve's Wine as your wine source but then our custom build system doesn't use the steam runtime and compiles against your bleeding edge libs instead, potentially introducing issues or changes in behavior, so that wouldn't be good enough anyway.

So in short,
for Proton-tkg: always report here.
for Wine-tkg: if you're not using the clean or staging profiles with `_LOCAL_PRESET`, always report here.

Depending on the issue, I might ask you to try with a clean preset before forwarding it to winehq.

**Comment by CartoonFan** [comment]
All right, sounds good.

**Comment by jnines** [comment]
May want to tweak the ReadMe as `git pull` doesn't actually pull everything, `git pull --recurse-submodules` does (or I have a complete fundamental misunderstanding of git, which is totally possible).

**Comment by Tk-Glitch** [comment]
If the clone was done with `--recurse-submodules`, a simple `git pull` should be enough. It definitely is on my end, at least, and a few users also confirmed it to work. Yet someone else has reported a similar problem. I'm not sure why it would work for some and not others.

**Comment by gardotd426** [comment]
@Tk-Glitch, so is there any way to make sure our changes to submodule config files don't get overwritten with every git pull? Because with the old method, if I edited, for example: `linux56-rc-tkg/customization.cfg`, as one does, and then ran `git add linux56-rc-tkg/customization.cfg && git commit && git pull` it would pull in the updates, but leave my changes intact. 

However, now it seems there's no way to do that, outside ditching the submodules altogether and cloning every submodule repo as it's own git repo individually. Because you can't add the changed files with from within the submodule, because you're not attached to any branch so `git pull` does just errors out, and it won't allow you to `git add` the changed files from the actual `PKGBUILDS` repo, because you get this: `fatal: Pathspec 'linux-tkg/linux56-tkg/PKGBUILD' is in submodule 'linux-tkg'`. But if you go into the (for example) `linux-tkg` submodule, and then run `git add linux56-tkg/customization.cfg`, then `git commit`, then drop back to the PKGBUILDS dir and run a git pull, the customized files get completely overwritten. 

If I'm not mistaken, it seems like this is just an inherent limitation and we will have to re-customize every single file every time we update, is this true?

**Comment by jnines** [comment]
@gardotd426 Look at `_EXT_CONFIG_PATH` at the top of the customization.cfg.  You can store the config files in `~/.config/frogminer/name.cfg` and it will override the default config.

**Comment by gardotd426** [comment]
@jnines I should have chosen another example file, because I always edit the PKGBUILD too, because I have a Navi card that suffers from [this bug](https://gitlab.freedesktop.org/drm/amd/-/issues/892), so I have to use a custom source for the kernel instead of the vanilla kernel.org source, among other things. So basically I'm asking about all other files, not just the ones that would be covered by `_EXT_CONFIG_PATH`.

**Comment by Tk-Glitch** [comment]
@gardotd426 In your case, `git stash` *might* work. I haven't experimented with it with submodules but I guess it's worth a shot.

**Comment by Tk-Glitch** [comment]
To get updates when using the PKGBUILDS repo, the recommended way is now to use the `TkgThingy` script at the root of the repo.
Closing :smile:
